src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -97,21 +97,20 @@ public ServiceEndpoint(string nameWithEndpointType, Uri endpoint, TokenCredentia
         /// <param name="credential"></param>
         /// <param name="endpointType"></param>
         /// <param name="name"></param>
-        public ServiceEndpoint(Uri endpoint, TokenCredential credential, EndpointType endpointType = EndpointType.Primary, string name = "")
+        public ServiceEndpoint(Uri endpoint, TokenCredential credential, EndpointType endpointType = EndpointType.Primary, string name = "") : this(endpoint)
         {
-            if (endpoint == null)
-            {
-                throw new ArgumentNullException(nameof(endpoint));
-            }
-            else if (endpoint.Scheme != Uri.UriSchemeHttp && endpoint.Scheme != Uri.UriSchemeHttps)
-            {
-                throw new ArgumentException("Endpoint scheme must be 'http://' or 'https://'");
-            }
             AccessKey = new AadAccessKey(endpoint, credential);
-            AudienceBaseUrl = BuildAudienceBaseUrl(endpoint);
-            Endpoint = BuildServerEndpoint(endpoint);
-            ClientEndpoint ??= Endpoint;
+            (Name, EndpointType) = (name, endpointType);
+        }
 
+        public ServiceEndpoint(string nameWithEndpointType, Uri endpoint, AzureKeyCredential credential) : this(endpoint, credential)
```

Comment:
In which case user pass in an AzureKeyCredential?

Reply:
supporting ConnectionString should be enough?

Reply:
I would like to follow the `WebPubSub` C# SDK. I though it may be a convention since it was developed by `Azure SDK Team`

Reply:
maybe not.. we are not following the azure sdk convention, and SignalR SDK is a little bit special that it is not only a REST client